intellij/test/junit/saros/intellij/editor/annotations/AnnotationManagerTest.java

```diff
@@ -144,12 +152,47 @@ public void testAddSelectionAnnotationEditor() throws Exception {
 
     prepareMockAddRemoveRangeHighlighters();
     mockAddSelectionRangeHighlighters(expectedRange);
+    mockAddCaretRangeHighlighters(end);
     replayMockAddRemoveRangeHighlighters();
 
     assertTrue(selectionAnnotationStore.getAnnotations().isEmpty());
 
     /* call to test */
-    annotationManager.addSelectionAnnotation(user, file, start, end, editor);
+    annotationManager.addSelectionAnnotation(user, file, start, end, editor, false);
+
+    /* check assertions */
+    List<SelectionAnnotation> selectionAnnotations = selectionAnnotationStore.getAnnotations();
+    assertEquals(1, selectionAnnotations.size());
+
+    SelectionAnnotation selectionAnnotation = selectionAnnotations.get(0);
+    assertAnnotationIntegrity(selectionAnnotation, user, file, expectedRange, editor);
+  }
+
+  /**
+   * Tests that the caret annotation is added to the front of the annotation for backwards
+   * selection.
+   *
+   * <p>The actual check of the annotation position is done implicitly by setting up the caret
```

Comment:
Thank you for this comment 👍 I was already searching for the position in the asserts.